docs/source/basic/introduction.rst

```diff
@@ -0,0 +1,18 @@
+Introduction
+============
+
+The basic concept of vikunja is to run an ``algorithm`` with an ``operator`` over a range of elements.
+
+* The ``algorithm`` specifies how the elements of one ore more input ranges are accessed and what kind of operator is applied on it. The algorithm also determines the shape of the result. Examples are:
+
+  * **Tranform**: Takes a range of elements as input and returns a range of the same size. **Transform** applies an operator on each element of the input range.
+  * **Reduce**: Takes a range of elements as input and returns a single element. The reduce operator takes two elements of the input range, applies an operation to them, and returns a single element. The operator is applied up to the point where only one element remains.
```

Comment:
I assume the reduction operations are expected to be associative, and probably also commutative, right? Otherwise, parallelization would be impossible with consistent results. Maybe add that here.

Reply:
That's correct and your are right, I should document it.